docs/Language-specification.md

```diff
@@ -732,6 +732,8 @@ Depends foo.o : foo.c ; # bang! can't build foo.c
 
 `foo.c` would no longer be a leaf source, and Ham would not know how to build it.
 
+TODO: The above example has errors in it, but I am not sure _exactly_ where. I suppose `Includes` needs to be replaced with `Depends`. But do we also need to replace `can't build foo.c` with `can't build foo.o`? Also, I don't get it. Why does foo.c need to be a leave source? Why is not generated.h becoming the leave source instead?
```

Comment:
Yes, it needed to be `Depends`, but the error actually was that `foo.c` can't be built.

Because `foo.c` has a dependency, Ham treats it as a _generated_ object instead of a source file. There is no action to build `foo.c` (because it shouldn't be built), so there is an error.

It's natural to think that `foo.c` depends on `foo.h` because it includes it, but the actual `foo.c` source file is independent of the header. When the header file changes, `foo.c` does not change, only things compiled using `foo.c` do. Hence `foo.h` is a dependency of `foo.o`, not `foo.c`.

I'll try to make this clearer, as this quirk is generally hidden by higher-level build systems.